server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -943,7 +947,7 @@ private SegmentIdWithShardSpec createNewSegment(
 
         return new SegmentIdWithShardSpec(
             dataSource,
-            overallMaxId.getInterval(),
+            interval,
```

Comment:
Changed to emphasize that the `interval` used in the `SegmentIdWithShardSpec` is always the same as the one passed to the method `createNewSegment`.